### PR TITLE
Feature/generic decoder testing

### PR DESCRIPTION
**Pull request by JosephFreshworks**
Added Intrumented Testing and Unit Testing to SHCDecoder library

### CODE REVIEW

File: shcDecoder/src/test/java/ca/bc/gov/shcdecoder/cache/CacheManagerImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        mockkStatic(Log::class)
+        sut = CacheManagerImpl(
+            config,
+            preferenceRepository,
+            fileManager
+        )
+    }
+
+    @Test
+    fun onFetch_cacheIsNotExpired_doNothing(): Unit = runBlocking {
```
Comment (bsavoini-fw) [inline shcDecoder/src/test/java/ca/bc/gov/shcdecoder/cache/CacheManagerImplTest.kt:51]: I'd suggest using the [GIVEN/WHEN/THEN style](https://martinfowler.com/bliki/GivenWhenThen.html) to improve readability. 
You can also [use backticks](https://kotlinlang.org/docs/coding-conventions.html#names-for-test-methods) in the method to replace the underscores by spaces (they are supported on /test/ only). 
For instance:
**\`GIVEN cache is fetched WHEN cache Is not expired THEN do nothing\`**

File: shcDecoder/build.gradle [hunk 2]
```diff
@@ -57,6 +57,8 @@ dependencies {
     testImplementation 'org.json:json:20210307'
 
     testImplementation "junit:junit:$versions.junit"
+    testImplementation "org.mockito:mockito-core:$versions.mockito"
+    testImplementation "io.mockk:mockk:$versions.mockk"
```
Comment (rashmi-bambhania) [inline shcDecoder/build.gradle:61]: mockk is wrapper of mockito. so do we really require mockk and mockito both?

Comment (JosephFreshworks) [reply shcDecoder/build.gradle:61]: This should be correct, however, if I remove Mockito Core dependency tests stop working somehow.

File: shcDecoder/src/test/java/ca/bc/gov/shcdecoder/cache/CacheManagerImplTest.kt (hunk 1 above)
Comment (JosephFreshworks) [reply shcDecoder/src/test/java/ca/bc/gov/shcdecoder/cache/CacheManagerImplTest.kt]: Done, thanks for the code convention advice.

**Review by bsavoini-fw — APPROVED** [review]

File: shcDecoder/src/androidTest/java/ca/bc/gov/shcdecoder/SHCVerifierImplTest.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
...82438630372754325762839456421225824680926317036044040334408766112280634102908097663033508055022776568664444323766396827723059290960005076242528060670383469440844625306597406080565236929557576676121686311763064555721522853413832057032032035394162660831095668595024324562685605407376405762620725761077322156726303505623533936632330557558035609645606230940550560583566712652046675325607200054624540276953120977442027576340760726277761352043617033112163636709225860615530702104547239557667624275681131340321337229401242302404502964203924456342566307284339253505692043757266413557505752657577337429441105543709272661302824364559772043444367336255314161372937425769586465623272625345063774672724286254066609572656554372433410076940563725404000220745742654645320660408226035264025097474047253307130743126370739047541684050300443666976245254703427563158455034676821716361706631324556043924272971110753116933256055270845317137665554761257117535566373094064503553724100103569397228557110751268566125500930572535057250325660564006395374016252660645212843592075501033007006357640103269773827707035385327442245370076773240693939123468412500574144555039656944755544436943343838565225521200377400760061653506263374"
```
Comment (pinakinkansara) [inline shcDecoder/src/androidTest/java/ca/bc/gov/shcdecoder/SHCVerifierImplTest.kt:83]: @JosephFreshworks 
Is it reasonable to create a helper class for all mock data for the test?

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: shcDecoder/src/androidTest/java/ca/bc/gov/shcdecoder/SHCVerifierImplTest.kt (hunk 3 above)
Comment (JosephFreshworks) [reply shcDecoder/src/androidTest/java/ca/bc/gov/shcdecoder/SHCVerifierImplTest.kt:83]: Could be, let me do it

**Review by will-freshworks — APPROVED** [review]

**Pull request merged**